lib/supavisor/protocol/server.ex

```diff
@@ -258,12 +265,15 @@ defmodule Supavisor.Protocol.Server do
     end
   end
 
-  @spec auth_request() :: iodata()
-  def auth_request() do
-    @auth_request
+  @spec auth_request(:scram | :md5) :: iodata
+  def auth_request(method \\ :scram, salt \\ nil) do
```

Comment:
I would split this in two functions, given the usage of default arguments and having one argument that only applies in one case:

    def md5_request(salt)
    def scram_request()

WDYT?

Reply:
sounds good 👍